primitives/authority-selection-inherents/src/authority_selection_inputs.rs

```diff
@@ -11,14 +11,33 @@ use {
 	main_chain_follower_api::DataSourceError, sidechain_domain::McEpochNumber,
 };
 
-/// The part of data for selection of authorities that comes from the main chain.
+/// The part of data for selection of authorities that comes from the main chains.
 /// It is unfiltered, so the selection algorithm should filter out invalid candidates.
+/// Note, this is part of the state storage (see pallet_session_validator_management).
 #[derive(Clone, Debug, Encode, Decode, TypeInfo, PartialEq, Eq)]
 pub struct AuthoritySelectionInputs {
 	pub d_parameter: DParameter,
+
+	/// List of permissioned candidates from the side chain, one for each sidechain_public_key.
 	pub permissioned_candidates: Vec<PermissionedCandidateData>,
+
+	// TODO ETH: The items in the Vec should be uniquely identified by sidechain_public_key
+	// (instead of  mainchain_public_key). This is necessary to allow one PC operator to register
+	// stake on different chains. The registration schema is symmetric with all chains and hence
+	// the same sidechain_public_key can be related to many public keys on different chains.
+	// Thus:
+	// - only one CandidateRegistrations per each candidate
+	// - each registered_candidates[i]:
+	//   - should contain valid mainchain_pub_key
```

Comment:
can you please clarify which `mainchain_pub_key`, ethereum? If so, do we need optional `eth_pub_key` as suggested bellow?

Reply:
clarified in the code comments.

Reply:
so in case of a valid eth candidate `mainchain_pub_key` == `Some(eth_pub_key)`?

Reply:
No, `mainchain_pub_key` is on Cardano, and `eth_pub_key` is on Ethereum, they can never be the same. The `eth_pub_key` field is optional, because `registered_candidates` vector should have only one item for each PC operator identified by `sidechain_public_key`, in other words, `registered_candidates` should have been `Map<SidechainPublicKey, CandidateRegistrations>`.

The reason I didn't introduced this map, is because I didn't want to deal with serialisation, but we can assume there is such mapping next to `registered_candidates`.